apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx

```diff
@@ -0,0 +1,174 @@
+import colors from '~/shared/theme/colors';
+
+import { CmsTypes } from '../cms_types';
+
+type FaqPairProps = {
+  content: CmsTypes.Faq;
+  isOpen: boolean;
+  onClick: () => void;
+};
+
+function FaqPair({ content, isOpen, onClick }: FaqPairProps) {
+  const contentRef = useRef<HTMLDivElement | null>(null);
+  const [contentHeight, setContentHeight] = useState(0);
+
+  useEffect(() => {
+    if (contentRef.current) {
+      setContentHeight(contentRef.current.scrollHeight);
+    }
+  }, [content]);
+
+  const toggleExpansion = useSpring({
+    opacity: isOpen ? 1 : 0,
+    height: isOpen ? `${contentHeight}px` : '0px',
+    config: { duration: 200, easing: (t) => t },
+  });
+
+  const rotatePlusIcon = useSpring({
+    transform: isOpen ? 'rotate(180deg)' : 'rotate(0deg)',
+    config: { duration: 200, easing: (t) => t },
+  });
+
+  const togglePlusIconOpacity = useSpring({
+    opacity: isOpen ? 0 : 1,
+    config: { duration: 200, easing: (t) => t },
+  });
+
+  return (
+    <StyledFaqPair isOpen={isOpen} onClick={onClick} justify="space-between">
+      <VStack>
+        <div>
+          <StyledQuestionText>{content.question}</StyledQuestionText>
+        </div>
+        <AnimatedAnswerContainer style={toggleExpansion} ref={contentRef}>
+          <StyledAnswerText>{content.answer}</StyledAnswerText>
+        </AnimatedAnswerContainer>
+      </VStack>
+      <Button style={rotatePlusIcon}>
+        <HorizontalLine />
+        <VerticalLine style={togglePlusIconOpacity} />
+      </Button>
+    </StyledFaqPair>
+  );
+}
+
+const AnimatedAnswerContainer = styled(animated.div)`
+  overflow: hidden;
+  box-sizing: border-box;
+`;
+
+const Line = styled(animated.div)`
+  position: absolute;
+  background-color: black;
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+const HorizontalLine = styled(Line)`
+  width: 100%;
+  height: 1px;
+  top: 50%;
+`;
+
+const VerticalLine = styled(Line)`
+  width: 1px;
+  height: 100%;
+  top: 0;
+`;
+
+const Button = styled(animated.div)`
+  margin-top: 8px;
+  position: relative;
+  width: 12px;
+  min-width: 12px;
+  height: 12px;
+  cursor: pointer;
+`;
+
+const StyledFaqPair = styled(HStack)<{ isOpen: boolean }>`
+  background-color: ${({ isOpen }) => (isOpen ? colors.offWhite : 'none')};
+  cursor: pointer;
+  transition: background-color 0.3s ease-in-out;
+  padding: 8px 16px;
+`;
+
+const StyledQuestionText = styled(TitleDiatypeL)`
+  font-weight: 400;
+
+  @media only screen and ${breakpoints.desktop} {
+    font-size: 20px;
+    line-height: 32px;
+  }
+`;
+
+const StyledAnswerText = styled(TitleDiatypeL)`
+  padding: 8px 0;
+  font-size: 16px;
+  line-height: 24px;
+  font-weight: 400;
+`;
+
+type Props = {
+  content: CmsTypes.FaqModule;
+};
+
+export default function Faq({ content }: Props) {
+  const [openIndex, setOpenIndex] = useState<number | null>(null);
+
+  const handleClick = (index: number) => {
+    setOpenIndex(openIndex === index ? null : index);
```

Comment:
nit: you should just be able to do `setOpenIndex(index)`. AFAIK react should know that nothing has changed

Reply:
sweet, simplified!

Reply:
nvm remembered I needed this in order to close a section that's currently open! so I added it back